bash/clean_sort_compress_data.sh

```diff
@@ -62,7 +62,7 @@ for file in $(find ${src_dir} -maxdepth 1 -iname 'omop_*.csv' -not -iname '*no_q
         # Change to Unix EOLs (delete "CR" from "CRLF"; `dos2unix`
         # doesn't work as a filter) and clean the fields.  Then sort by
         # patient ID.
-        ${timer_cmd} sed -e 's/\r//g' ${sub_dblqt} ${file} | ${timer_cmd} awk -F , -f ${script_dir}/clean_data.awk | ${timer_cmd} sort --stable --field-separator=, --buffer-size=10G --temporary-directory=${TMPDIR} --key=${key} > ${dst_file}
+        ${timer_cmd} sed -e 's/\r//g' ${sub_dblqt} ${file} | ${timer_cmd} awk -F , -f ${script_dir}/clean_data.awk --re-interval | ${timer_cmd} sort --stable --field-separator=, --buffer-size=10G --temporary-directory=${TMPDIR} --key=${key} > ${dst_file}
```

Comment:
Could you add to the above comment something about needing `--re-interval` to support older `awk`s and what version you are running that needs this option? (BTW, what does `awk --version` print for you?)

Also, as a matter of style, I tend to put options without arguments before options with arguments, and I think that would be a good idea here as well.

Reply:
Also, please put your name as an author in the copyright notice. :smiley:

Reply:
I pushed an updated file with your suggested changes.  The version of Awk on the quasar machine I was using is GNU Awk 3.1.5.  It looks like the interval syntax with {} is a posix standard, but earlier versions( <4.0) of awk required the --re-interval flag.  I think it was an attempt to prevent older awk scripts from breaking.  I'm hoping versions >= 4.0 just ignore the flag?
Also, I did add an additional sorting key so that files are sorted by patient_id first, then by *_occurrence_id.  This ends up giving events sorted by date for each pt as well.  Thought it might be useful.